libs/core/src/lib/date-picker/date-picker.component.ts

```diff
@@ -529,6 +534,11 @@ export class DatePickerComponent<D>
             const calendar = this._calendars.first;
             this._calendarComponent = calendar;
             setTimeout(() => {
+                if (this._calendarComponent) {
+                    this._focusTrapService?.pauseCurrentFocusTrap();
```

Comment:
This seems to work well but I am wondering if we specifically want to look for calendar open/close changes and not any change to _calendars?

Reply:
The only change current behaviour provides is either there is a calendar, or there's no calendar, so we are safe (unless we decide to add more calendars to one popover for some reason :))